QMigrate/signal/local_mag/amplitude.py

```diff
@@ -0,0 +1,894 @@
+        ----------
+        tr : `obspy.Trace` object
+            Trace to be filtered
+
+        Returns
+        -------
+        filter_sos : ndarray
+            Second-order sections representation of the applied filter.
+
+        """
+
+        # Check if we have been diverted here from trying to apply a bandpass
+        # filter. Else use specified params for highpass.
+        if self.bandpass_filter:
+            filt_freq = self.bandpass_lowcut
+        else:
+            filt_freq = self.highpass_freq
+        corners = self.filter_corners
+
+        f_nyquist = 0.5 * tr.stats.sampling_rate
+        f_crit = filt_freq / f_nyquist
+
+        # Pre-process and apply filter
+        tr.detrend("linear")
+        tr.taper(0.05, "cosine")
+        tr.filter(type="highpass", freq=filt_freq, corners=corners,
+                  zerophase=False)
+
+        # Generate filter coefficients for the highpass filter we
+        # applied; this is how the filter is designed within ObsPy
+        filter_sos = iirfilter(N=corners, Wn=f_crit, btype="highpass",
+                               ftype="butter", output="sos")
+
+        return filter_sos
+
+    def _get_amplitude_windows(self, station, i, event, p_ttimes, s_ttimes):
+        """
+        Calculate the start and end time of the windows to measure the max P-
+        and S- wave amplitudes in. This is done on the basis of the pick times,
+        the event marginal window, the traveltime and uncertainty and the
+        specified S-wave signal window.
+
+        P_window_start : P_pick - marginal_window - traveltime_uncertainty
+        P_window_end : equivalent to start, or S_pick time; whichever is
+                       earlier
+        S_window_start : same as P
+        S_window_end : S_pick + signal_window + marginal_window +
+                       traveltime_uncertainty
+
+        traveltime_uncertainty = traveltime * fraction_tt
+            (where fraction_tt is as specified for the autopicker).
+
+        Parameters
+        ----------
+        station : str
+            Station name.
+        i : int
+            Iterator variable.
+        event : `QMigrate.io.Event` object
+            Light class encapsulating signal, onset, pick and location
+            information for a given event.
+        p_ttimes : array-like
+            Array of interpolated P traveltimes to the requested grid position.
+        s_ttimes : array-like
+            Array of interpolated S traveltimes to the requested grid position.
+
+        Returns
+        -------
+        windows : array-like
+            [[P_window_start, P_window_end], [S_window_start, S_window_end]]
+        picked : bool
+            Whether at least one of the phases was picked by the autopicker.
+
+        """
+
+        p_pick, s_pick, picked = self._get_picks(i, event)
+
+        fraction_tt = event.picks["fraction_tt"]
+
+        # Check p_pick is before s_pick
+        assert p_pick < s_pick # NOTE: No catch for this
```

Comment:
Yep I added code to deal with the `AssertionError` while bug testing. Left it like this so that while I was testing it would clearly flag this.

It's the kind of bug where it's difficult to decide whether it's best to make it break the code, or handle it quietly. If this situation has occurred then something has gone pretty wrong.